src/algorithms/tracking/CKFTracking.cc

```diff
@@ -74,10 +79,53 @@ namespace eicrecon {
         std::unique_ptr<edm4eic::TrackParametersCollection>,
         std::vector<ActsExamples::Trajectories*>
     >
-    CKFTracking::process(const ActsExamples::IndexSourceLinkContainer &src_links,
-                         const ActsExamples::MeasurementContainer &measurements,
+    CKFTracking::process(const edm4eic::Measurement2DCollection& meas2Ds,
                          const edm4eic::TrackParametersCollection &init_trk_params) {
 
+        // auto src_links    = std::vector<std::shared_ptr<ActsExamples::IndexSourceLink>>();
+
+        // create sourcelink and measurement containers
+        auto measurements = std::make_shared<ActsExamples::MeasurementContainer>();
+
+          // need list here for stable addresses
+        std::list<ActsExamples::IndexSourceLink> sourceLinkStorage;
+        ActsExamples::IndexSourceLinkContainer src_links;
+        src_links.reserve(meas2Ds.size());
+        std::size_t  hit_index = 0;
+
+
+        for (const auto& meas2D : meas2Ds) {
+            // --Create source links - method of trackersourcelinkerrResult
+            // auto sourceLink = std::make_shared<ActsExamples::IndexSourceLink>(meas2D.getSurface(), hit_index);
+            // src_links.emplace_hint(src_links.end(), *sourceLink);
+
+            // --follow example from ACTS
+            sourceLinkStorage.emplace_back(meas2D.getSurface(), hit_index);
+            ActsExamples::IndexSourceLink& sourceLink = sourceLinkStorage.back();
+            // Add to output containers:
+            // index map and source link container are geometry-ordered.
+            // since the input is also geometry-ordered, new items can
+            // be added at the end.
+            src_links.insert(src_links.end(), sourceLink);
+            // ---
+            // Create ACTS measurements
+            Acts::Vector2 loc = Acts::Vector2::Zero();
+            auto pos_x = meas2D.getLoc().a;
+            auto pos_y = meas2D.getLoc().b;
+            loc[Acts::eBoundLoc0] = pos_x;
+            loc[Acts::eBoundLoc1] = pos_y;
+
+
+            Acts::SymMatrix2 cov = Acts::SymMatrix2::Zero();
+            cov(0, 0) = meas2D.getCovariance().xx;
+            cov(1, 1) = meas2D.getCovariance().yy;
```

Comment:
We should also add cross-covariances.

Reply:
The off diagonal elements are all zero. We can probably add them in the future when we know what number to use

Reply:
You should add them now here, even if they are zero.